code/modules/projectiles/guns/misc/beam_rifle.dm

```diff
@@ -16,6 +16,8 @@
 	icon = 'icons/obj/guns/energy.dmi'
 	icon_state = "esniper"
 	inhand_icon_state = "esniper"
+	worn_icon_state = null
+	worn_icon = 'icons/mob/clothing/back.dmi'

```

Comment:
this is already handled by `build_worn_icon` via the slot flag, remove it.

Reply:
Reverted. Had the worn_icon set to make the other slots use a working icon. I'll add it to the list of missing textures.